File: Server/U-Tracker API/src/main/kotlin/com/code_of_duty/utracker_api/data/models/Cycle.kt
package com.code_of_duty.utracker_api.data.models

import com.code_of_duty.utracker_api.data.enums.CycleType
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "cycle")
data class Cycle(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: UUID = UUID.randomUUID(),
    val cycleType: CycleType,
    val name: String,
    @Column(name = "order_value")
    val orderValue: Int? = 1,
    @ManyToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "pensum_fk", referencedColumnName = "id")
    val pensum: Pensum,
    @ManyToMany(cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    @JoinTable(
        name = "subject_x_cycle",
        joinColumns = [JoinColumn(name = "cycle_fk", referencedColumnName = "id")],
        inverseJoinColumns = [JoinColumn(name = "subject_fk", referencedColumnName = "code")]
    )
    val subjects: List<Subject> = emptyList()
)

File: Client/Website/src/interfaces/Cycle.ts
import {Pensum} from "@/interfaces/Pensum";
import {Subject} from "@/interfaces/Subject";

export interface Cycle {
    id?: string,
    type: number,
    name: string,
    pensum: Pensum,
}

export type CyclePreview = {pensumId: string} & Omit<Cycle, "pensum">;

export type CycleResponse = {
    "cycleType": string,
    "orderValue": number,
    "subjects": Subject[],
} & Omit<Cycle, 'pensum' | 'type'>

type Assessments = {
    id: string,
    name: string,
    percentage: number,
    date: string,
    grade: number,
}